src/character_attire.cpp

```diff
@@ -2230,6 +2230,61 @@ void outfit::pickup_stash( const item &newit, int &remaining_charges, bool ignor
     }
 }
 
+static std::vector<pocket_data_with_parent> get_child_pocket_with_parent(
+    const item_pocket *pocket, const item_location &parent, item_location it, const int nested_level )
+{
+    std::vector<pocket_data_with_parent> ret;
+    if( pocket != nullptr ) {
+        pocket_data_with_parent pocket_data = { pocket, item_location::nowhere, nested_level };
+        const item_location new_parent = item_location( it );
+
+        if( parent != item_location::nowhere ) {
+            pocket_data.parent = item_location( parent, it.get_item() );
+        }
+
+        for( const item *contained : pocket->all_items_top() ) {
+            const item_location poc_loc = item_location( it, const_cast<item *>( contained ) );
+            for( const item_pocket *pocket_nest : contained->get_all_contained_pockets() ) {
+                std::vector<pocket_data_with_parent> child =
+                    get_child_pocket_with_parent( pocket_nest, new_parent,
+                                                  poc_loc, nested_level + 1 );
+                for( const pocket_data_with_parent &pock_d : child ) {
+                    ret.emplace_back( pock_d );
+                }
+            }
+        }
+        ret.emplace_back( pocket_data );
+    }
+    return ret;
+}
+
+std::vector<pocket_data_with_parent> outfit::get_all_pocket_with_parent(
```

Comment:
Since this is checking worn and wielded items, it should be `Character` scope, not outfit. I suggest moving the sorting and filtering here, so it's kind of like an advanced `best_pocket`.

Reply:
Thank you for code review.
I agree with your view in the main. Though, I feel that it is better to make the another function for sort.
Not sure if it will work, but I will try.

Reply:
If you want to keep it flexible, you can implement the sorting (and filtering) as predicate parameters. As long as this isn't used anywhere else, it's not strictly necessary, though.

Reply:
Okay.
I am still learning modern programming, but I would like to give it a try.

Reply:
I followed your advice and changed the scope and moved the filtering and sorting method to the function side.
Readability and code flexibility seem to have improved.

Reply:
I think it's fine where it is, since it's about character equipment.